### PR TITLE
Arbitrary tater models

### PR DESCRIPTION
**Pull request by eutropius225**
This PR adds support for any model to be used for a tiny potato. Models of the form `botania:tiny_potato/<name>` will automatically be loaded, and will appear when a tiny potato's name, (lowercased and with `[ -]` replaced by `_`) matches `<name>`.

For example, it allows for the `pluraltater|manytater|systemtater|systater` to be added without code changes.

This PR reimplements the existing `kyle hyde` and `aureylion` taters, as well as all those added in #3654, using the new system, and makes the halloween tater available at any time as `halloween|spooky|spookytater`.

taters after
They looked like this before
taters before

### CODE REVIEW

File: src/main/resources/assets/botania/models/block/shapes/tiny_potato.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "parent": "minecraft:block/block",
+  "elements": [
```
Comment (williewillus) [inline src/main/resources/assets/botania/models/block/shapes/tiny_potato.json:3]: why not just set "particle": "#all" here so you don't have to worry about particle in any of the code or generated models above?

**Comment by williewillus** [comment]
also some of the taters in your screenshot don't match, on the front row.

**Comment by eutropius225** [comment]
> also some of the taters in your screenshot don't match, on the front row.

Third from the left was erroneously named `egg` instead of `eggtater`, the inconsistency has been fixed.
Grumm on the far right lost its name, that's been fixed too.
Botater's sides were on the wrong way around, that's been fixed.
(Far left is `spooky`, which is new)
ss

File: src/main/resources/assets/botania/models/block/shapes/tiny_potato.json (hunk 1 above)
Comment (eutropius225) [reply src/main/resources/assets/botania/models/block/shapes/tiny_potato.json]: It's only really relevant for the `default` model. In any case, all of the models now inherit from `block/cube_all` which does have `particle` bound to `#all`.

**Comment by williewillus** [comment]
pog

**Pull request merged**